clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleServerGroupCachingAgentTest.java

```diff
@@ -710,6 +715,101 @@ void serverGroupAutoscalerProperties() {
     assertThat(serverGroup.getAutoscalingMessages()).containsExactly("message1", "message2");
   }
 
+  @Test
+  void serverGroupAutoscalingPolicy_allFields() {
+
+    AutoscalingPolicy input =
+        new AutoscalingPolicy()
+            .setCoolDownPeriodSec(123)
+            .setCpuUtilization(new AutoscalingPolicyCpuUtilization().setUtilizationTarget(9.87))
+            .setLoadBalancingUtilization(
+                new AutoscalingPolicyLoadBalancingUtilization().setUtilizationTarget(6.54))
+            .setMaxNumReplicas(99)
+            .setMinNumReplicas(11)
+            .setMode("ON")
+            .setCustomMetricUtilizations(
+                ImmutableList.of(
+                    new AutoscalingPolicyCustomMetricUtilization()
+                        .setMetric("myMetric")
+                        .setUtilizationTarget(911.23)
+                        .setUtilizationTargetType("GAUGE"),
+                    new AutoscalingPolicyCustomMetricUtilization()));
+
+    InstanceGroupManager instanceGroupManager =
+        new InstanceGroupManager().setName("myServerGroup").setZone(ZONE_URL);
+    Autoscaler autoscaler =
+        new Autoscaler().setZone(ZONE_URL).setTarget("myServerGroup").setAutoscalingPolicy(input);
+
+    Compute compute =
+        new StubComputeFactory()
+            .setInstanceGroupManagers(instanceGroupManager)
+            .setAutoscalers(autoscaler)
+            .create();
+    AbstractGoogleServerGroupCachingAgent cachingAgent =
+        createCachingAgent(
+            compute, ImmutableList.of(instanceGroupManager), ImmutableList.of(autoscaler));
+
+    CacheResult cacheResult = cachingAgent.loadData(inMemoryProviderCache());
+    GoogleServerGroup serverGroup = getOnlyServerGroup(cacheResult);
+    GoogleAutoscalingPolicy converted = serverGroup.getAutoscalingPolicy();
+
+    assertThat(converted.getCoolDownPeriodSec()).isEqualTo(input.getCoolDownPeriodSec());
+    assertThat(converted.getCpuUtilization().getUtilizationTarget())
+        .isEqualTo(input.getCpuUtilization().getUtilizationTarget());
+    assertThat(converted.getLoadBalancingUtilization().getUtilizationTarget())
+        .isEqualTo(input.getLoadBalancingUtilization().getUtilizationTarget());
+    assertThat(converted.getMaxNumReplicas()).isEqualTo(input.getMaxNumReplicas());
+    assertThat(converted.getMinNumReplicas()).isEqualTo(input.getMinNumReplicas());
+    assertThat(converted.getMode().toString()).isEqualTo(input.getMode());
+
+    assertThat(converted.getCustomMetricUtilizations())
+        .hasSize(input.getCustomMetricUtilizations().size());
+    for (int i = 0; i < converted.getCustomMetricUtilizations().size(); ++i) {
+      CustomMetricUtilization convertedCustomMetric =
+          converted.getCustomMetricUtilizations().get(0);
```

Comment:
Should `.get(0)` be `.get(i)` here and on the next line?

Reply:
Whoops, fixed, thanks!